packages/shared/scope-hoisting/src/link.js

```diff
@@ -323,7 +323,10 @@ export function link({
             path.replaceWith(
               THROW_TEMPLATE({MODULE: t.stringLiteral(source.value)})
             );
-          } else if (dep.isWeak && !bundle.env.isLibrary) {
+          } else if (
+            dep.isWeak &&
+            !bundle.env.isLibrary /* TODO dep.wasDeferred*/
```

Comment:
Not happy with using `isLibrary` here, adding something like `excludeNodeModules` could break this again in the future.
Should really be `if(dep.wasDeferred)`